packages/@aws-cdk/aws-apprunner/lib/vpc-ingress-connection.ts

```diff
@@ -0,0 +1,161 @@
+
+  /**
+   * The domain name associated with the VPC Ingress Connection resource.
+   * @attribute
+   */
+  readonly domainName: string;
+
+  /**
+   * The current status of the VPC Ingress Connection.
+   * @attribute
+   */
+  readonly status: string;
+
+  /**
+   * The name of the VPC Ingress Connection.
+   * @attribute
+   */
+  readonly vpcIngressConnectionName: string;
+}
+
+/**
+ * Represents the App Runner VPC Ingress Connection.
+ */
+export interface IVpcIngressConnection extends cdk.IResource {
+  /**
+   * The Amazon Resource Name (ARN) of the VPC Ingress Connection.
+   * @attribute
+   */
+  readonly vpcIngressConnectionArn: string;
+
+  /**
+    * The domain name associated with the VPC Ingress Connection resource.
+    * @attribute
+    */
+  readonly domainName: string;
+
+  /**
+    * The current status of the VPC Ingress Connection.
+    * @attribute
+    */
+  readonly status: string;
+
+  /**
+    * The name of the VPC Ingress Connection.
+    * @attribute
+    */
+  readonly vpcIngressConnectionName: string;
+}
+
+/**
+ * The App Runner VPC Ingress Connection
+ *
+ * @resource AWS::AppRunner::VpcIngressConnection
+ */
+export class VpcIngressConnection extends cdk.Resource implements IVpcIngressConnection {
+  /**
+   * Import from VPC Ingress Connection attributes.
+   */
+  public static fromVpcIngressConnectionAttributes(scope: Construct, id: string, attrs: VpcIngressConnectionAttributes): IVpcIngressConnection {
+    const vpcIngressConnectionArn = attrs.vpcIngressConnectionArn;
+    const domainName = attrs.domainName;
+    const status = attrs.status;
+    const vpcIngressConnectionName = attrs.vpcIngressConnectionName;
+
+    class Import extends cdk.Resource {
+      public readonly vpcIngressConnectionArn = vpcIngressConnectionArn;
+      public readonly domainName = domainName;
+      public readonly status = status;
+      public readonly vpcIngressConnectionName = vpcIngressConnectionName;
+    }
+
+    return new Import(scope, id);
+  }
+
+  /**
+   * The Amazon Resource Name (ARN) of the VPC Ingress Connection.
+   * @attribute
+   */
+  readonly vpcIngressConnectionArn: string;
+
+  /**
+   * The domain name associated with the VPC Ingress Connection resource.
+   * @attribute
+   */
+  readonly domainName: string;
+
+  /**
+   * The current status of the VPC Ingress Connection.
+   * @attribute
+   */
+  readonly status: string;
+
+  /**
+   * The name of the VPC Ingress Connection.
+   * @attribute
+   */
+  readonly vpcIngressConnectionName: string;
+
+  public constructor(scope: Construct, id: string, props: VpcIngressConnectionProps) {
+    super(scope, id, {
+      physicalName: props.vpcIngressConnectionName,
+    });
+
+    const resource = new CfnVpcIngressConnection(this, 'Resource', {
+      ingressVpcConfiguration: {
+        vpcEndpointId: props.vpcInterfaceEndpoint.vpcEndpointId,
+        vpcId: props.vpc.vpcId,
+      },
+      serviceArn: props.service.serviceArn,
```

Comment:
```suggestion
  public constructor(scope: Construct, id: string, { vpcIngressConnectionName, vpc: { vpcId }, service: { serviceArn }, vpcInterfaceEndpoint: { vpcEndpointId } }: VpcIngressConnectionProps) {
    super(scope, id, {
      physicalName: vpcIngressConnectionName,
    });

    const resource = new CfnVpcIngressConnection(this, 'Resource', {
      ingressVpcConfiguration: {
        vpcEndpointId,
        vpcId
      },
      serviceArn
```

Reply:
Thanks for the suggestion. While I appreciate the suggestion, I can't find any other places in the code base where this code style is taken.